### PR TITLE
Electron cleanup

### PR DESCRIPTION
**Pull request by cldwalker**
This PR makes our Electron app easier to maintain as follows:
* No more outdated references to atom-shell (except for one we can't avoid without upgrading)
* Cleaned up disjointed electron app. Parts of the app were in two different directories with two package.jsons. Much cleaner and simpler to reason about
* Added an electron guide
* Moved platform and shell under deploy as being top-level was for node-webkit

I QAed building and using it but the more the merrier. Also added some misc doc cleanup and improvements. If you're using Reminisce, the rename will break its behavior. Use [this branch](https://github.com/cldwalker/Reminisce/tree/electron) in the meantime and I will submit a PR as soon as this branch lands.

@kenny-evitt and @rundis Would you find it helpful if I pull in [this commit](https://github.com/cldwalker/LightTable/commit/00da32118f2565414f2e4be8c12510c78681c65d)? I used this a number of times while developing the electron app. It helped to understand the flow of ipc messages and to visualize subtle timing issues with messages

After this PR, I'm ready to focus on the release :)

### CODE REVIEW

**Comment by kenny-evitt** [comment]
@cldwalker This looks great. But I just ran into #1958 trying to build this locally. I'm going to look into that. Or should I just wait for the Electron version bump? That will supposedly fix it.

**Comment by cldwalker** [comment]
@kenny-evitt Pushed a commit that fixes it and relevant script changes. @rundis There's no longer the issue I mentioned of the LT icon being crossed in osx's system tray. Turns out I hadn't renamed Atom to Electron in Info.plist and renaming it fixed that issue.
I still have to add io.js directions and the debugging commit but hope to get this PR ready soon

**Comment by kenny-evitt** [comment]
@cldwalker The build worked. And I was able to run a Clojure project.

I did get an error logged to the console when I started; see below. But I had accidentally opened two Light Table instances. When I closed both and then re-opened just one, I didn't see the error.

```
SyntaxError: Unexpected end of input
    at Object.parse (native)
    at file:///home/kenny/@code/LightTable/builds/lighttable-0.8.0-linux/resources/app/core/node_modules/lighttable/bootstrap.js:35320:223
    at file:///home/kenny/@code/LightTable/builds/lighttable-0.8.0-linux/resources/app/core/node_modules/lighttable/bootstrap.js:23148:14
    at null.<anonymous> (file:///home/kenny/@code/LightTable/builds/lighttable-0.8.0-linux/resources/app/core/node_modules/lighttable/bootstrap.js:23160:16)
    at goog.events.Listener.handleEvent (file:///home/kenny/@code/LightTable/builds/lighttable-0.8.0-linux/resources/app/core/node_modules/lighttable/bootstrap.js:18552:51)
    at Object.goog.events.fireListener (file:///home/kenny/@code/LightTable/builds/lighttable-0.8.0-linux/resources/app/core/node_modules/lighttable/bootstrap.js:18787:12)
    at Object.goog.events.fireListeners_ (file:///home/kenny/@code/LightTable/builds/lighttable-0.8.0-linux/resources/app/core/node_modules/lighttable/bootstrap.js:18777:53)
    at Object.goog.events.dispatchEvent (file:///home/kenny/@code/LightTable/builds/lighttable-0.8.0-linux/resources/app/core/node_modules/lighttable/bootstrap.js:18830:47)
    at goog.events.EventTarget.dispatchEvent (file:///home/kenny/@code/LightTable/builds/lighttable-0.8.0-linux/resources/app/core/node_modules/lighttable/bootstrap.js:18951:22)
    at goog.net.XhrIo.dispatchErrors_ (file:///home/kenny/@code/LightTable/builds/lighttable-0.8.0-linux/resources/app/core/node_modules/lighttable/bootstrap.js:20786:62)
```

[I think the build failure was due to two recent releases of Electron bumping the version we were using to the 'second page' of results from the GitHub API.](https://github.com/LightTable/LightTable/issues/1958#issuecomment-136239010)

**Comment by rundis** [comment]
First try seems promising. Will try it at work tomorrow and report if I find anything of importance.

**Comment by cldwalker** [comment]
Added $IPC_DEBUG. [Should look this](https://www.dropbox.com/s/xixn2r38j9lz5qk/Screenshot%202015-08-31%2018.00.54.png?dl=0) in the terminal. LT can only log the renderer process but terminal can see main and renderer process

**Comment by cldwalker** [comment]
@rundis @kenny-evitt This is ready to go per what we talked about Sunday. Please use it and let me know if you come across any major issues. It's been solid for me so far

**Comment by rundis** [comment]
I'm getting an electron icon in the taskbar on my Mac

**Comment by cldwalker** [comment]
@rundis I only see this with `script/light.sh` which is expected. Are you seeing this with script/build.sh and then calling builds/*/light?

**Comment by rundis** [comment]
@cldwalker My bad, I forgot you mentioned that.
I haven't had any problems with this branch so far, so it seems good to me.

**Comment by cldwalker** [comment]
@kenny-evitt The latest working for you? Any objection to merging?

File: deploy/core/version.json [hunk 1]
```diff
@@ -1 +1 @@
-{"version":"0.7.2","atom-shell":"0.19.5"}
+{"version":"0.7.2","electron":"0.19.5"}
```
Comment (rundis) [inline deploy/core/version.json]: Should the electron version correspond with the one in deploy/electron/Gruntfile.js ?

Comment (cldwalker) [inline deploy/core/version.json]: Makes sense. Will do

**Comment by kenny-evitt** [comment]
@cldwalker This seems fine based on my *very* limited tests so far. Clojure works.

**Comment by cldwalker** [comment]
Ok. Merging. We can downgrade electron if we come across any blockersin QA

**Pull request merged**